### DISCUSSION TITLE
Ansible and Terraform runners

### DISCUSSION THREAD

**Discussion by erjac77** [nautobot/nautobot#996] [Ideas]
Following the first Nautobot Community Meeting two weeks ago, I'm looking to start a discussion on creating Ansible and Terraform local runners for jobs. Maybe something similar to the Nornir plugin, so other plugins and jobs can be built on top of it.

https://github.com/ansible/ansible-runner

One use-case in mind: We want to use the Ansible collection for Arista Validated Designs (AVD) to configure Arista devices. A local Ansible runner would be helpful.
https://github.com/aristanetworks/ansible-avd

**Comment by jvanderaa** [comment]
I've been _starting_ to think a small bit about the build of an `execution environment` into this list. I think that those would be hosted over on https://github.com/nautobot/nautobot-ansible/.

**Reply by brobare** [reply]
Would this lock it into AWX/Tower?

**Reply by smk4664** [reply]
I think an Ansible Execution Environment could be nice for https://github.com/nautobot/nautobot-ansible/. But the ask looks to be a way to call Ansible within Nautobot Jobs.  This could possibly look similar to https://github.com/nautobot/nautobot-plugin-nornir.

**Reply by jvanderaa** [reply]
Got it. That makes sense. You should be able to import any package into a Job. In this example https://josh-v.com/nautobot-jobs-execution/ I imported the Meraki library.

**Reply by jifox** [reply]
@erjac77 Thanks for referencing my discussion entry [Utilize ansible-runner to execute playbooks from jobs/api/...](https://github.com/nautobot/nautobot/discussions/1281) to this discussion.

Is there any activity already going on?

**Reply by smk4664** [reply]
@jifox I started to brainstorm on this, with no real work yet. I wouldn't mind trying to jot down some ideas.

Would you be opposed to me creating a repo for this and tracking the discussion and brainstorming there?

**Reply by jifox** [reply]
@smk4664 Thanks for your offer to create a repo. I appreciate this.

**Reply by jpedersen01** [reply]
Did this move forward?  Running ansible playbooks from Nautobot would be a huge benefit.
